web-app/src/main/webapp/script.js

```diff
@@ -70,17 +75,65 @@ function showReportForm() {
 }
 
 async function loadPoliceReports(map) {
+  // Clear all markers on the map
+  for (const marker of mapMarkers) {
+    marker.setMap(null);
+  }
+  mapMarkers = [];
+
   const FILE_NAMES = ['2019_12_london', '2020_01_london', '2020_02_london', '2020_03_london', '2020_04_london', '2020_05_london']
-  for (const file_name of FILE_NAMES) {
-    const data = await fetch('../data/' + file_name + '.json');
-    const reports = await data.json();
-    for (report of reports) {
-      new google.maps.Marker({
-        position: {
-          lat: Number(report.latitude),
-          lng: Number(report.longitude)
-        }, map: map
-      });
-    };
+  const uncheckedCategoriesElement = Array.from(document.querySelectorAll("input.category:not(:checked)"));
+  const uncheckedCategories = uncheckedCategoriesElement.map(element => element.value);
+  const numberOfMonths = Number(document.querySelector("input.time-frame:checked").value);
+
+  mapMarkers = await Promise.all(FILE_NAMES.map((file_name) =>
```

Comment:
nit: maybe just create a new local variable for this part. Reusing variables can be confusing because the name of the variable no longer describes what the variable is.